main.go

```diff
@@ -3,116 +3,61 @@ package main
+	err := sync(ctx)
 	if err != nil {
-		_, _ = fmt.Fprintf(os.Stderr, "%v\n", err)
+		fmt.Fprintf(os.Stderr, "ERROR: %v\n", err)
 		os.Exit(1)
 	}
+	os.Exit(0)
 }
 
-func newApp() (context.Context, context.CancelFunc, *cli.App) {
-	logInstance := &atomic.Value{}
-	logInstance.Store(logr.Discard())
-	app := &cli.App{
-		Name:     appName,
-		Usage:    appLongName,
-		Version:  fmt.Sprintf("%s, revision=%s, date=%s", version, commit, date),
-		Compiled: compilationDate(),
-
-		EnableBashCompletion: true,
-
-		Before: setupLogging,
-		Flags: []cli.Flag{
-			&cli.BoolFlag{
-				Name:    "debug",
-				Aliases: []string{"verbose", "d"},
-				Usage:   "sets the log level to debug",
-				EnvVars: envVars("DEBUG"),
-			},
-			&cli.StringFlag{
-				Name:        "log-format",
-				Usage:       "sets the log format (values: [json, console])",
-				EnvVars:     envVars("LOG_FORMAT"),
-				DefaultText: "console",
-			},
-		},
-		Commands: []*cli.Command{
-			newExampleCommand(),
-		},
-		ExitErrHandler: func(ctx *cli.Context, err error) {
-			if err != nil {
-				AppLogger(ctx).Error(err, "fatal error")
-				cli.HandleExitCoder(cli.Exit("", 1))
-			}
-		},
+func cfg() (string, string) {
+	exoscaleApiKey := os.Getenv(keyEnvVariable)
+	if exoscaleApiKey == "" {
+		fmt.Fprintf(os.Stderr, "ERROR: Environment variable %s must be set\n", keyEnvVariable)
+		os.Exit(1)
 	}
-	hasSubcommands := len(app.Commands) > 0
-	app.Action = rootAction(hasSubcommands)
-	// There is logr.NewContext(...) which returns a context that carries the logger instance.
-	// However, since we are configuring and replacing this logger after starting up and parsing the flags,
-	// we'll store a thread-safe atomic reference.
-	parentCtx := context.WithValue(context.Background(), loggerContextKey{}, logInstance)
-	ctx, stop := signal.NotifyContext(parentCtx, syscall.SIGINT, syscall.SIGTERM)
-	return ctx, stop, app
-}
 
-func rootAction(hasSubcommands bool) func(context *cli.Context) error {
-	return func(ctx *cli.Context) error {
-		if hasSubcommands {
-			return cli.ShowAppHelp(ctx)
-		}
-		return LogMetadata(ctx)
+	exoscaleApiSecret := os.Getenv(secretEnvVariable)
+	if exoscaleApiSecret == "" {
+		fmt.Fprintf(os.Stderr, "ERROR: Environment variable %s must be set\n", secretEnvVariable)
+		os.Exit(1)
 	}
-}
 
-// env combines envPrefix with given suffix delimited by underscore.
-func env(suffix string) string {
-	return envPrefix + "_" + suffix
+	return exoscaleApiKey, exoscaleApiSecret
 }
 
-// envVars combines envPrefix with each given suffix delimited by underscore.
-func envVars(suffixes ...string) []string {
-	arr := make([]string, len(suffixes))
-	for i := range suffixes {
-		arr[i] = env(suffixes[i])
-	}
-	return arr
-}
+func sync(ctx context.Context) error {
+	exoscaleApiKey, exoscaleApiSecret := cfg()
 
-func compilationDate() time.Time {
-	compiled, err := time.Parse(time.RFC3339, date)
+	ex := egoscale.NewClient("https://api-ch-gva-2.exoscale.com", exoscaleApiKey, exoscaleApiSecret)
```

Comment:
Maybe also add an env var for the URL?

Reply:
For now I see no reason why this would need to be configurable...